src/peep/staff.c

```diff
@@ -327,6 +327,20 @@ void game_command_hire_new_staff_member(int* eax, int* ebx, int* ecx, int* edx,
 									   edi);
 }
 
+void game_command_callback_hire_new_staff_member(int eax, int ebx, int ecx, int edx, int esi, int edi, int ebp)
+{
+    int sprite_index = edi;
+    if (sprite_index == 0xFFFF)
+    {
+        rct_window *window = window_find_by_class(WC_STAFF_LIST);
+        window_invalidate(window);
+    }
+    else
+    {
+        window_staff_open(&get_sprite(sprite_index)->peep);
```

Comment:
I encourage that you pull expressions out into locals as it makes it much easier to debug and diagnose problems.
```
rct_peep * peep = &get_sprite(sprite_index)->peep;
window_staff_open(peep);
```